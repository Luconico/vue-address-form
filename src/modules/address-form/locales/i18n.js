import { createI18n } from 'vue-i18n';
import datetimeFormats from './datetimeFormats';
import numberFormats from './numberFormats';
import * as es from './es.json';
import * as en from './en.json';
import * as it from './it.json';
import * as pt from './pt.json';
import * as fr from './fr.json';

const i18n = createI18n({
  datetimeFormats,
  numberFormats,
  locale: window.location.pathname.split('/')[1],
  messages: {
    es: { ...es },
    en: { ...en },
    it: { ...it },
    pt: { ...pt },
    fr: { ...fr },
  },
});

export default i18n;
