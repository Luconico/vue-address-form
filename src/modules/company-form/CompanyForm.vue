<template>
  <template v-for="(field, index) in companyForm.fields" :key="index">
    <InputBuilder
      v-if="field.needed"
      :type="field.type"
      :disabled="isDisabled"
      :label="$t(index)"
      :required="field.require"
      v-model="field.value"
      :name="index"
      :customClass="{ 'custom-form-group': true, '--error': field.error }"
    />
  </template>
</template>

<script>
import InputBuilder from "@/shared/forms/input-builder/InputBuilder.vue";
import useCompanyForm from "./composables/useCompanyForm";

export default {
  name: "CompanyForm",
  components: {
    InputBuilder,
  },
  setup() {
    const { companyForm, isDisabled, isFetching, selectOptions } = useCompanyForm();
    return {
      isDisabled,
      isFetching,
      companyForm,
      selectOptions,
    };
  },
};
</script>

<style>
</style>
