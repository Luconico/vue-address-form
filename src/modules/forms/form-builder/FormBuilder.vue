<template>
  <AlertMessages v-if="messages.length > 0" :messages="messages" />
  <form :id="customId" :class="customClass" @submit.prevent="onSumbit">
    <slot />
  </form>
</template>

<script>
import { ref } from "vue";
import AlertMessages from "@/shared/alert-message/AlertMessages.vue";
import {  onBeforeMount, onMounted } from "@vue/runtime-core";
import { useStore } from "vuex";
import dtxApi from "@/api/dtxApi";
export default {
  name: "FormBuilder",
  components: {
    AlertMessages,
  },
  emits: ["onSubmit"],
  props: {
    customId: {
      type: String,
      default: "form",
    },
    customClass: {
      type: Object,
      default: () => ({}),
    },
    initialValues: {
      type: Object,
      default: null,
    },
    url: {
      type: String,
      default: null,
    },
    successMessage: {
      type: String,
      default: 'Form submitted successfully',
    },
    errorMessage: {
      type: String,
      default: 'Error submitting form',
    },
  },
  setup(props, { slots, emit }) {
    const store = useStore();
    const formValues = ref({});
    const messages = ref([]);

    onBeforeMount(() => {
      if (props.initialValues) store.dispatch("formBuilder/initialValues", props.initialValues)
    });

    onMounted(() => {
      registerFormModules();
    });

    const registerFormModules = () => {
      slots.default().map((slot) => {
        if (slot.type.name.includes("Form"))
          store.dispatch("formBuilder/pressentModule", slot.type.name.toLowerCase().replace("form", ""));
      });
    };

    return {
      messages,
      onSumbit: () => {
        formValues.value = {};
  
        store.getters['formBuilder/pressentModules'].map((module) => {
          store.dispatch(`${module}Form/submit`);
          const { fields } = store.getters[`${module}Form/formValues`];
          formValues.value = { ...formValues.value, ...fields };
        });

        store.dispatch("formBuilder/isSubmitting", true);
        console.log(formValues);
        dtxApi
          .post(props.url, formValues)
          .then((response) => {
            console.log(response);
            store.dispatch("formBuilder/isSubmitting", false);
            store.dispatch("formBuilder/isSubmited", true);
            messages.value = [{ msgType: "success", value: props.successMessage }];
            emit("onSubmit", {ok: true, data: response.data});
          })
          .catch((error) => {
            console.log(error);
            store.dispatch("formBuilder/isSubmitting", false);
            messages.value = [{ msgType: "error", value: props.errorMessage }];
            emit("onSubmit", {ok: false, data: error});
          });
      },
    };
  },
};
</script>

<style>
input:focus-visible,
textarea:focus-visible {
  outline: none;
}


.suggestions-container,
.error-message-container {
  width: 100%;
  position: relative;
}

.suggestions-list {
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 100%;
  left: 0;
  width: 100%;
  background-color: #fff;
  border: 1px solid #ccc;
  border-top: none;
  border-radius: 0 0 5px 5px;
  box-shadow: inset 0px -1px 5px 0px #ccc;
  overflow: hidden;
  z-index: 1;
}

.suggestions-list > span {
  padding: 5px;
  cursor: pointer;
  text-align: left;
}

.suggestions-list > span:hover {
  background-color: #ccc;
}

.error-message {
  position: absolute;
  font-size: 0.8rem;
  font-style: italic;
}

.masked-container {
  position: relative;
}
.masked-value {
  position: absolute;
  pointer-events: none;
  top: 4px;
  left: 2px;
  width: 95%;
  background-color: #fff;
  overflow: hidden;
  z-index: 1;
  padding-left: 2px;
  padding-bottom: 1px;
}
input:disabled,
select:disabled,
textarea:disabled,
.masked-disabled {
  cursor: not-allowed;
}
.masked-disabled {
  background-color: #F8F8F8;
  pointer-events: auto;
}
</style>