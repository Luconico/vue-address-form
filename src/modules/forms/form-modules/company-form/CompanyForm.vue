<template>
  <template v-for="(field, index) in companyForm.fields" :key="index">
    <InputBuilder
      v-if="field.active"
      :type="field.type"
      :disabled="isDisabled"
      :label="$t(index)"
      :required="field.validations.includes('required')"
      v-model="field.value"
      :name="index"
      :validations="field.validations"
      :customClass="[customClass, {'--error': field.error}]"
      @onValidated="onValidated"
    />
  </template>
</template>

<script>
import InputBuilder from "@/modules/forms/input-builder/InputBuilder.vue";
import useCompanyForm from "./composables/useCompanyForm";

export default {
  name: "CompanyForm",
  components: {
    InputBuilder,
  },
   props: {
    customClass: {
      type: String,
      default: "",
    },
  },
  setup() {
    const { companyForm, isDisabled, onValidated } = useCompanyForm();
    return {
      isDisabled,
      companyForm,
      onValidated
    };
  },
};
</script>

<style>
</style>
