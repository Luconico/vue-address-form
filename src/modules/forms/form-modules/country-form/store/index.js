
import state from './state'
import * as actions from './actions'
import * as getters from './getters'
import * as mutations from './mutations'


const countryFormModule = {
    namespaced: true,
    actions,
    getters,
    mutations,
    state
}


export default countryFormModule
