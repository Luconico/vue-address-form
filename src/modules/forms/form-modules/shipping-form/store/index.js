
import state from './state'
import * as actions from './actions'
import * as getters from './getters'
import * as mutations from './mutations'


const shippingFormModule = {
    namespaced: true,
    actions,
    getters,
    mutations,
    state
}


export default shippingFormModule
