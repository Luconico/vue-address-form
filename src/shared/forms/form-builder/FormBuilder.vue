<template>
  <AlertMessages v-if="messages.length > 0" :messages="messages" />
  <form :id="customId" :class="customClass" @submit.prevent="null">
    <slot name="inputs" />
  </form>
</template>

<script>
import AlertMessages from "@/shared/alert-message/AlertMessage.vue";
export default {
  name: "form-builder",
  components: {
    AlertMessages,
  },
  props: {
    messages: {
      type: Array,
      default: () => [],
    },
    customId: {
      type: String,
      default: "form",
    },
    customClass: {
      type: Object,
      default: () => ({}),
    },
  }
};
</script>

<style>
</style>