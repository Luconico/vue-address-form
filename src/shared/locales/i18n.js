import { createI18n } from 'vue-i18n';
import datetimeFormats from './datetimeFormats';
import numberFormats from './numberFormats';
import * as es from './es.json';
import * as en from './en.json';
import * as it from './it.json';
import * as pt from './pt.json';
import * as fr from './fr.json';
import { LOCATION } from '@/global';

const i18n = createI18n({
  datetimeFormats,
  numberFormats,
  locale: LOCATION,
  messages: {
    es: { ...es },
    en: { ...en },
    it: { ...it },
    pt: { ...pt },
    fr: { ...fr },
  },
});

export default i18n;
