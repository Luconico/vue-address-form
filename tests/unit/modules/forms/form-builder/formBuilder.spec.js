import FormBuilder from '@/modules/forms/form-builder/FormBuilder.vue'
import { mount } from '@vue/test-utils'
import store from '@/store'
import i18n from '@/shared/locales/i18n'
import CompanyForm from '@/modules/forms/form-modules/company-form/CompanyForm.vue';
const maska = jest.fn()

describe('FormBuilder.vue', () => {

    const wrapper = mount(FormBuilder, {
        global: {
            plugins: [store, i18n],
            directives: {
                maska
            }
        },
        slots: {
            default: CompanyForm
        },

    })

    it('should render the component', () => {
        expect(wrapper.exists()).toBe(true)
    })

})