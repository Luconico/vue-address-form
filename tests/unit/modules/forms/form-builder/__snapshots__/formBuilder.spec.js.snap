// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBuilder.vue should match with snapshot 1`] = `
<!--v-if-->
<form id="testId" class="custom-form">
  <div class=""><span>companyName <span title="msg.required">*</span>
    <!--v-if--></span>
    <!--v-if--><input name="companyName" type="text">
    <div class="suggestions-container">
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
  <div class=""><span>cif <span title="msg.required">*</span>
    <!--v-if--></span>
    <div class="masked-container">
      <div class="masked-value">test</div>
    </div><input name="cif" type="text">
    <div class="suggestions-container">
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</form>
`;
